### PR TITLE
update for V2 Staking contracts

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description

Added changes to support the new staking contracts

About # (link your issue here)

https://github.com/GoodDollar/GoodProtocolUI/issues/133

# How Has This Been Tested?

Locally, on production, through the metaStake debug functions (stakeList only!)
For review: I couldn't test the actual making of stakes and/or withdrawing rewards

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/pages/gd/Stake/index.tsx [hunk 1]
```diff
@@ -202,14 +205,17 @@ const StakeTable = ({
                                                 setActiveStake(stake)
                                                 setActiveTableName()
                                             }}
-                                        >
-                                            {i18n._(t`Stake`)}
+                                        > { stake.isV2 || stake.protocol === LIQUIDITY_PROTOCOL.GOODDAO ?
```
Comment (sirpy) [inline src/pages/gd/Stake/index.tsx:208]: this should be same condition as the "disabled" on line 198. just so it is more clear

File: src/pages/gd/Stake/index.tsx [hunk 2]
```diff
@@ -218,8 +224,11 @@ const StakeTable = ({
...
```
Comment (sirpy) [inline src/pages/gd/Stake/index.tsx:227]: same

File: src/sdk/contracts/SimpleStakingContract.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ export function simpleStakingContract(web3: Web3, address: string) {
  * @returns {Promise<string[]>}
  */
 export async function getSimpleStakingContractAddresses(web3: Web3): Promise<string[]> {
-    const chainId = await getChainId(web3)
+    const chainId = await getChainId(web3) // doesn't return correct chainId
```
Comment (sirpy) [inline src/sdk/contracts/SimpleStakingContract.ts:24]: what?

File: src/sdk/contracts/SimpleStakingContractV2.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  //     case SupportedChainId.FUSE:
+  //         deploymentName = CURRENT_NETWORK
+  //         break
+  // }
+
+  if (protocol === LIQUIDITY_PROTOCOL.COMPOUND){
+    // TODO: when contracts are released on testnet use deploymentName here
+    usdOracle = contractsAddresses['production-mainnet'].compound.daiUsdOracle
+  } else {
+    usdOracle = contractsAddresses['production-mainnet'].aave.usdcUsdOracle
```
Comment (sirpy) [inline src/sdk/contracts/SimpleStakingContractV2.ts:70]: need to consider NETWORK
ropsten, kovan. keys: kovan-mainnet=kovan, staging-mainnet/fuse-mainnet = ropsteen

**Review by sirpy — CHANGES_REQUESTED** [review]

File: src/sdk/contracts/SimpleStakingContract.ts (hunk 3 above)
Comment (L03TJ3) [reply src/sdk/contracts/SimpleStakingContract.ts:24]: getChainId for some reason always returns chainID 1
which doesn't make sense if you have switched network
it doesn't make sense in any way, as the underlying function is nothing more then web3.eth.getChainId

I don't know the reason... yet

File: src/sdk/contracts/SimpleStakingContractV2.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    deploymentName = 'production-mainnet'    
+  } else if (chainId == SupportedChainId.ROPSTEN && CURRENT_NETWORK == 'fuse') {
+    deploymentName = 'fuse-mainnet'
+  } else if (chainId == SupportedChainId.ROPSTEN && CURRENT_NETWORK == 'staging') {
+    deploymentName = 'staging-mainnet'
+  } else if (chainId == SupportedChainId.KOVAN) {
+      deploymentName = 'kovan-mainnet'
+  } else {
+      deploymentName = 'production-mainnet'
+  }
```
Comment (sirpy) [inline src/sdk/contracts/SimpleStakingContractV2.ts:66]:
```suggestion
  deploymentName = 'production-mainnet'
  if (chainId == SupportedChainId.MAINNET || chainId == SupportedChainId.ROPSTEN) {
    deploymentName = `${CURRENT_NETWORK}-mainnet`    
  } else if (chainId == SupportedChainId.KOVAN) {
      deploymentName = 'kovan-mainnet'
  }
```

**Pull request merged**